eth/stagedsync/stage_bor_heimdall.go

```diff
@@ -86,6 +96,31 @@ func BorHeimdallForward(
 	} else {
 		headNumber, err = stages.GetStageProgress(tx, stages.Headers)
 
+		hash, err := cfg.blockReader.CanonicalHash(ctx, tx, headNumber)
```

Comment:
We need to have this code segment even in case when it is mining, milestones helps us detect that it is on a fork and will rewind it to last verified canonical block inhibiting large reorgs.

Reply:
I have added changes to the stage code so that mining is also handled.